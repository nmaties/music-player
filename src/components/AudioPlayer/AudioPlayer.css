.audio-visualizer{
    background:  url(https://media.giphy.com/media/4HtL2Lu70WgdJeTICr/giphy.gif) no-repeat no-repeat center center;
    background-size: cover;
    height: 22px;
    display: inline-block;
    width: 130px;
}

.filled-icon {
    color: darkorange;
}

.audio-duration{
}

.audio-controls > span {
    margin-right: 5px;
}

.slider {
    margin-top: 14px;
    -webkit-appearance: none;
    height: 10px;
    background: orange;
    outline: none;
    opacity: 0.7;
    -webkit-transition: .2s;
    transition: opacity .2s;
}

.slider:hover {
    opacity: 1;
}

.slider::-webkit-slider-thumb {
    -webkit-appearance: none;
    appearance: none;
    width: 15px;
    height: 15px;
    background: darkorange;
    cursor: pointer;
}

.slider::-moz-range-thumb {
    width: 15px;
    height: 15px;
    background: darkorange;
    cursor: pointer;
}
