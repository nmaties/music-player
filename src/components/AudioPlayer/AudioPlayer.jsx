import React, { Component } from 'react';
import './AudioPlayer.css';

class AudioPlayer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      play: false,
      duration: 0,
      interval: null
    };

    this.player = React.createRef();
  }

  formatCurrentTime = (time) => new Date(time * 1000).toISOString().substr(11, 8);

  getCurrentDuration = () => {
    this.setState({
      duration: this.player.current.currentTime
    })
  };

  play = () => {
    this.setState({
      play: true,
      interval: setInterval(this.getCurrentDuration, 1000)
    });
    this.player.current.play();
  };

  pause = () => {
    clearInterval(this.state.interval);
    this.setState({
      play: false,
      interval: null
    });
    this.player.current.pause();
  };

  setVolume = (evt) => {
    let nodeVal = evt.target.value / 100;
    this.player.current.volume = nodeVal;
  };

  render() {
    return (
        <div>
          <audio ref={this.player}>
            <source src={this.props.song.songLink}/>
          </audio>
          <div className="col-md-12 row">
            <div className="audio-controls col-md-6">
          <span className="btn btn-default">
            <i className={`fas fa-play ${this.state.play ? 'filled-icon' : ''}`}
               onClick={this.play}></i>
          </span>
              <span className="btn btn-default">
            <i className={`fas fa-pause ${!this.state.play ? 'filled-icon' : ''}`}
               onClick={this.pause}></i>
          </span>
            </div>
            <div className="col-md-6">
          <span>
              <input type="range" className="slider" onChange={this.setVolume}
                     name="volume" min="0" max="100" step="1"/>
          </span>
            </div>
          </div>
          {this.state.play &&
          <div className="col-md-12 row">
            <div className="col-md-6">{this.formatCurrentTime(this.state.duration)}</div>
            <div className="audio-duration col-md-6 audio-visualizer"></div>
          </div>}
        </div>
    );
  }
}

export default AudioPlayer;
