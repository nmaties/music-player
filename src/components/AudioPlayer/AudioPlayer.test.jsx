import React from 'react';
import renderer from 'react-test-renderer';
import AudioPlayer from './AudioPlayer';
import { mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('AudioPlayer', () => {

  const playerProps = {
    artist: 'Calum Scott',
    songName: 'You Are The Reason',
    format: '.mp3',
    releaseYear: 2018,
    album: 'Only Human',
    songLink: '1GYwj2eL61SkOOEsGakyTgNpong37WCWv',
    id: 1
  };

  describe('Snapshots', () => {
    it('is displayed fine', () => {
      const player = renderer
        .create(<AudioPlayer song = { playerProps } />);
      expect(player.toJSON()).toMatchSnapshot();
    });

    it('is displayed fine playing state', () => {
      const player = renderer
        .create(<AudioPlayer song = { playerProps } />);

      player.getInstance().setState({
        play: true,
        duration: 75
      });
      expect(player.toJSON()).toMatchSnapshot();
    });

    it('is displayed fine paused state', () => {
      const player = renderer
        .create(<AudioPlayer song = { playerProps } />);

      player.getInstance().setState({
        play: false,
        duration: 100
      });
      expect(player.toJSON()).toMatchSnapshot();
    });
  });

  describe('AudioPlayer functionality', ()=> {
    const player = mount(<AudioPlayer song = { playerProps } />);

    it('play button', () => {
      player.find('.fa-play').simulate('click');
      expect(player.state().play).toBeTruthy();
      expect(player.state().interval).not.toBeNull();
    });

    it('pause button', () => {
      player.find('.fa-pause').simulate('click');
      expect(player.state().play).toBeFalsy();
      expect(player.state().interval).toBeNull();
    });
  });
});