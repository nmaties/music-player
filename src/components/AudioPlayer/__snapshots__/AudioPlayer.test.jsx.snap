// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer Snapshots is displayed fine 1`] = `
<div>
  <audio>
    <source
      src="1GYwj2eL61SkOOEsGakyTgNpong37WCWv"
    />
  </audio>
  <div
    className="col-md-12 row"
  >
    <div
      className="audio-controls col-md-6"
    >
      <span
        className="btn btn-default"
      >
        <i
          className="fas fa-play "
          onClick={[Function]}
        />
      </span>
      <span
        className="btn btn-default"
      >
        <i
          className="fas fa-pause filled-icon"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="col-md-6"
    >
      <span>
        <input
          className="slider"
          max="100"
          min="0"
          name="volume"
          onChange={[Function]}
          step="1"
          type="range"
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`AudioPlayer Snapshots is displayed fine paused state 1`] = `
<div>
  <audio>
    <source
      src="1GYwj2eL61SkOOEsGakyTgNpong37WCWv"
    />
  </audio>
  <div
    className="col-md-12 row"
  >
    <div
      className="audio-controls col-md-6"
    >
      <span
        className="btn btn-default"
      >
        <i
          className="fas fa-play "
          onClick={[Function]}
        />
      </span>
      <span
        className="btn btn-default"
      >
        <i
          className="fas fa-pause filled-icon"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="col-md-6"
    >
      <span>
        <input
          className="slider"
          max="100"
          min="0"
          name="volume"
          onChange={[Function]}
          step="1"
          type="range"
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`AudioPlayer Snapshots is displayed fine playing state 1`] = `
<div>
  <audio>
    <source
      src="1GYwj2eL61SkOOEsGakyTgNpong37WCWv"
    />
  </audio>
  <div
    className="col-md-12 row"
  >
    <div
      className="audio-controls col-md-6"
    >
      <span
        className="btn btn-default"
      >
        <i
          className="fas fa-play filled-icon"
          onClick={[Function]}
        />
      </span>
      <span
        className="btn btn-default"
      >
        <i
          className="fas fa-pause "
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="col-md-6"
    >
      <span>
        <input
          className="slider"
          max="100"
          min="0"
          name="volume"
          onChange={[Function]}
          step="1"
          type="range"
        />
      </span>
    </div>
  </div>
  <div
    className="col-md-12 row"
  >
    <div
      className="col-md-6"
    >
      00:01:15
    </div>
    <div
      className="audio-duration col-md-6 audio-visualizer"
    />
  </div>
</div>
`;
